### DISCUSSION TITLE
Neutral (zero charge) particle's momentum get updated by field in 2D simulation

### DISCUSSION THREAD

**Discussion by GivenChen** [BLAST-WarpX/warpx#4589] [General]
Hi, I am not sure if it is a bug... I found it very strange that in my 2D simulations, all neutral particles are pushed by the grid field(as if they have +e charge), but actually, they are not ionized and have zero charge. Here is my simulation result: I am simulating an electron beam(initialized between **-100cm to 0cm**) passing through a region full of neutral hydrogen gas(initialized from **0cm to 70cm**).The hydrogen gas particles are not ionized but somehow pushed by the electric field from the beam.

The hydrogen species charge distribution: It is zero everywhere---meaning there is no ionization
![image1](https://github.com/ECP-WarpX/WarpX/assets/52523185/a3e5118a-a6dc-426a-8e18-795e4682fe96)

The average hydrogen momentum distribution(particle field ux of hydrogen species): They gain momentum somehow

![image2](https://github.com/ECP-WarpX/WarpX/assets/52523185/ceeaea66-df3c-4ebf-81c7-80172e8233b4)

I am very confused so I checked the source file, it looks like this line controls the ionizable particle's charge:
![image](https://github.com/ECP-WarpX/WarpX/assets/52523185/7f7c3960-1156-4504-9fa8-fc625af2f2d3)
So I did a little test and changed the line. I subtracted the ionization level by 1 here
![image](https://github.com/ECP-WarpX/WarpX/assets/52523185/c29307bc-3d14-46b5-908e-a566e209a500)
The result is simulation result didn't change at all.. So my guess is that ion_lev is not updated and remains null pointer?

Any help would be highly appreciated! I will attach my input file in the following as a txt file, and the Warpx version is the latest release 24.01.

[inputs_2d.txt](https://github.com/ECP-WarpX/WarpX/files/13863719/inputs_2d.txt)

**Comment by ax3l** [comment]
Hm, this seems to be a bug. Let's move this to an issue:
https://github.com/ECP-WarpX/WarpX/issues/4592
